e2e/__tests__/readInitialOptions.test.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import path = require('path');
+import execa = require('execa');
+import type {ReadJestConfigOptions, readInitialOptions} from 'jest-config';
+
+function resolveFixture(...pathSegments: Array<string>) {
+  return path.resolve(__dirname, '..', 'read-initial-options', ...pathSegments);
+}
+
+interface ProxyReadJestConfigOptions extends ReadJestConfigOptions {
+  cwd?: string;
+}
+
+/**
+ * These e2e tests are running via a child process, because we're running in a VM and are not allowed to `import` directly
+ * It also represents a more real-world example of how to run.
+ */
+async function proxyReadInitialOptions(
+  configFile?: string | undefined,
+  options?: ProxyReadJestConfigOptions,
```

Comment:
```suggestion
  configFile: string | undefined,
  options: ProxyReadJestConfigOptions,
```

Reply:
This would require all options, which they shouldn't be, right?

Reply:
this is just an internal helper in this test, and you always pass both. Attempt to apply this and see no type errors 🙂

Reply:
Ah yes, you're right. Sorry about that, I guess it was early this morning 😅